### PR TITLE
Add Dockerfile + Makefile for running common tasks

### PR DESCRIPTION
**Pull request by wesrice**
I've added a Dockerfile and Makefile that enables running benchmarks, running tests, and updating dependencies via a Docker container. The goal is to provide a basic, portable environment that a variety of developers can quickly get up and running.

### CODE REVIEW

File: README.dev.md [hunk 1]
```diff
@@ -1,17 +1,37 @@
...
-3. Review `releasenotes.md` for completeness and correctness. Update its release
+1. Review `releasenotes.md` for completeness and correctness. Update its release
    date.
-4. Run dev-bin/release.sh. This will build the project, generate docs, upload to
+1. Run dev-bin/release.sh. This will build the project, generate docs, upload to
    NuGet, and make a GitHub release.
-5. Update GitHub Release page for the release.
-6. Verify the release on [NuGet](https://www.nuget.org/packages/MaxMind.GeoIP2/).
+1. Update GitHub Release page for the release.
+1. Verify the release on [NuGet](https://www.nuget.org/packages/MaxMind.GeoIP2/).
```
Comment (oschwald) [inline README.dev.md]: Could we change this back to be sequentially ordered? Markdown parsers will do the right thing either way, but I suspect most people reading this are going to be reading the text file in their local repo.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+		--rm \
+		-v $$(pwd)/$(1):/project \
+		-e "CONFIGURATION=$(CONFIGURATION)" \
+		-e "CONSOLE_FRAMEWORK=$(CONSOLE_FRAMEWORK)" \
+		-e "DOTNETCORE=$(DOTNETCORE)" \
+		-e "MAXMIND_BENCHMARK_DB=$(MAXMIND_BENCHMARK_DB)" \
+		geoip2-dotnet \
+		/bin/sh -c "dotnet run \
+			-c \$(DOLLAR)CONFIGURATION \
+			-f \$(DOLLAR)CONSOLE_FRAMEWORK \
```
Comment (oschwald) [inline Makefile]: Why is `$(DOLLAR)` needed?

File: README.dev.md (hunk 1 above)
Comment (wesrice) [reply README.dev.md]: Updated and force pushed.

File: Makefile (hunk 2 above)
Comment (wesrice) [reply Makefile]: I had to use it to circumvent the way Make evaluates `$`. 

In this circumstance, both `$CONFIGURATION` and `$CONSOLE_FRAMEWORK` are part of a command that is run within the Docker container, making them strings that require the `$` to be a string. `$` is evaluated by Make. `$$` would bypass the evaluation in Make, but would be evaluated as part of the command. I also attempted to use `$$$`, `$$$$`, `\$$$`, and `\$$$$`, but they were either invalid or evaluated wrongly in the Docker container.  So I ended up assigning `$$` to a variable ensured that Make would perceive `$` as a string when evaluating the `DOLLAR` variable. The preceding backslash was needed for the command to evaluate properly in Docker.

Comment (oschwald) [reply Makefile]: Why do you need to circumvent it? With:

```
--- a/Makefile
+++ b/Makefile
@@ -23,9 +23,9 @@ benchmark:
                -e "DOTNETCORE=$(DOTNETCORE)" \
                -e "MAXMIND_BENCHMARK_DB=$(MAXMIND_BENCHMARK_DB)" \
                geoip2-dotnet \
-               /bin/sh -c "dotnet run \
-                       -c \$(DOLLAR)CONFIGURATION \
-                       -f \$(DOLLAR)CONSOLE_FRAMEWORK \
+               /bin/sh -c -x "dotnet run \
+                       -c $(CONFIGURATION) \
+                       -f $(CONSOLE_FRAMEWORK) \
                        -p ./MaxMind.GeoIP2.Benchmark/MaxMind.GeoIP2.Benchmark.csproj"
 
 build-image:
```

I see `+ dotnet run -c Release -f netcoreapp2.0 -p ./MaxMind.GeoIP2.Benchmark/MaxMind.GeoIP2.Benchmark.csproj` being run.

Comment (wesrice) [reply Makefile]: Ah, I see. Given that, I can probably avoid passing those two variables as environment variables. I'll clean this up and make another push.

**Pull request merged**